src/art.rs

```diff
@@ -46,8 +67,8 @@ impl fmt::Display for TrieError {
 // From the specification: Adaptive Radix tries consist of two types of nodes:
 // Inner nodes, which map prefix(prefix) keys to other nodes,
 // and leaf nodes, which store the values corresponding to the key.
-struct Node<P: Prefix + Clone, V: Clone> {
-    pub node_type: NodeType<P, V>, // Type of the node
+pub struct Node<P: Prefix + Clone, V: Clone> {
```

Comment:
I know this isnt related to the change, but do we maybe want `Copy` instead of `Clone`? Then its up to the structs passed in to reuse memory correctly?

Reply:
Thanks for raising the point, I'll need to have a quick look into the difference between Copy/Clone

Reply:
Thanks for bringing this up. I hadn't paid much attention to the Copy trait before. Based on what I've gathered, the Copy trait is suitable for types that exhibit a simple, bitwise copy behavior, which seems to apply to primitive types. However, if the value is a non-primitive type, I'm not sure if using Copy would work.

This discussion is quite interesting. For now, I'm considering sticking with implementing the Clone trait. I'll keep an eye on this topic for any future developments or insights